stevensStringLib.h

```diff
@@ -1148,8 +1148,39 @@ namespace stevensStringLib
         return std::equal(str.begin(), str.begin() + str.size()/2, str.rbegin());
     }
 
-
-    //replace
+    /**
+    * Replaces characters using a vector syntax. Use multi_type_vector to construct proper input.
+    * Example syntax:
+    * 
+    * std::string testing = stevensStringLib::replace_characters({"testing {0}{1}", "1", "This is another test"}); // replaces string selection with specified characters
+    * std::cout << testing << std::endl;
+    * 
+    * @param vector<str> - example_vector above
+    * @retval str - returns replaced string
+    */
+    std::string replace_characters( std::vector<std::string> values ) 
```

Comment:
I have a `replace` function in mind for the future, so I think I'm going to rename this function `format` for the purpose of reserving the function name `replace` for later, as well as the benefit of people having preloaded knowledge of the formatting library, so they can have some expectations of what this function does.